log: [refactor] Use info level for init logs

maflcko opened the pull request:
Many of the normal, and expected init logs, which are run once after startup use the deprecated alias of `LogInfo`.

Fix that by using `LogInfo` directly, which is a refactor, except for a few log lines that also have `__func__` removed.

(Also remove the unused trailing `\n` char while touching those logs)

fanquake commented on src/node/chainstate.cpp:
```diff
@@ -145,18 +145,19 @@ ChainstateLoadResult LoadChainstate(ChainstateManager& chainman, const CacheSize
-    LogPrintf("Setting nMinimumChainWork=%s\n", chainman.MinimumChainWork().GetHex());
+    LogInfo("Setting nMinimumChainWork=%s", chainman.MinimumChainWork().GetHex());
     if (chainman.MinimumChainWork() < UintToArith256(chainman.GetConsensus().nMinimumChainWork)) {
         LogPrintf("Warning: nMinimumChainWork set below default value of %s\n", chainman.GetConsensus().nMinimumChainWork.GetHex());
```
LogWarning?

stickies-v commented on src/wallet/init.cpp:
```diff
@@ -88,14 +88,14 @@ bool WalletInit::ParameterInteraction() const
         for (const std::string& wallet : gArgs.GetArgs("-wallet")) {
-            LogPrintf("%s: parameter interaction: -disablewallet -> ignoring -wallet=%s\n", __func__, wallet);
+            LogInfo("Parameter interaction: -disablewallet -> ignoring -wallet=%s", wallet);
         }
 
         return true;
     }
 
     if (gArgs.GetBoolArg("-blocksonly", DEFAULT_BLOCKSONLY) && gArgs.SoftSetBoolArg("-walletbroadcast", false)) {
-        LogPrintf("%s: parameter interaction: -blocksonly=1 -> setting -walletbroadcast=0\n", __func__);
+        LogInfo("Parameter interaction: -blocksonly=1 -> setting -walletbroadcast=0");
```
I think `LogWarning` could be appropriate here?

stickies-v replied:
Woops, sorry. I thought these parameter interactions handled incompatible explicit user settings, but they're only updating unset settings, in which case `LogInfo` is absolutely appropriate.

stickies-v commented on src/init/common.cpp:
```diff
@@ -115,9 +115,9 @@ bool StartLogging(const ArgsManager& args)
     }
 
     if (!LogInstance().m_log_timestamps)
```
nit: would be good to add braces or move to one-liner while touching this

stickies-v reviewed (approved):
ACK fa65716a878dbc0daabf1b31e832db5b230c285c

Mostly refactor operations, with a couple of minor logging statement updates (e.g. removing function names, adding detail on which operation is started/exited, ...)

l0rinc commented on src/node/blockstorage.cpp:
```diff
@@ -1232,17 +1232,17 @@ void ImportBlocks(ChainstateManager& chainman, std::span<const fs::path> import_
             chainman.LoadExternalBlockFile(file, &pos, &blocks_with_unknown_parent);
             if (chainman.m_interrupt) {
-                LogPrintf("Interrupt requested. Exit %s\n", __func__);
+                LogInfo("Interrupt requested. Exit reindexing.");
```
Would it be possible to separate the `LogPrintf` inlining from the `__func__` and rewording changes into focused commits?

l0rinc commented on src/init.cpp:
```diff
@@ -1275,18 +1277,18 @@ static ChainstateLoadResult InitAndLoadChainstate(
             index->Stop();
             if (!(index->Init() && index->StartBackgroundSync())) {
-                LogPrintf("[snapshot] WARNING failed to restart index %s on snapshot chain\n", index->GetName());
+                LogInfo("[snapshot] WARNING failed to restart index %s on snapshot chain", index->GetName());
```
I personally would find it simpler if we inlined every single `LogPrintf` call here, removed the [alias](https://github.com/bitcoin/bitcoin/blob/master/src/logging.h#L369-L370) and the [dedicated test](https://github.com/bitcoin/bitcoin/blob/master/src/test/logging_tests.cpp#L131), and do the info/warn/error differentiation in a separate PR.
I know that means we'd be touching the same lines multiple times, but in that case this PR could simply be a scripted diff with alias removal (+ line ending and __func__ and spacing fixes in separate commits) instead of having to check if we consider the text a warning or an info as well (since there are tens of remaining `LogPrintf` calls that I would still consider info logs but weren't migrated here for some reason.

l0rinc reviewed (approved):
Concept ACK, thanks for the cleanup.

I personally would prefer doing every single `LogPrintf` -> `LogInfo` inlining here in a scripted diff (so we can get rid of the alias), with follow-up commits fixing the line endings and formatting and stuff.

And I agree that the warn/error split should be done in a separate PR where we can focus on the context (we could probably do a scripted diff for most of the lines that contain warning/error to cover most of those as well)

fanquake commented:
> I personally would prefer doing every single LogPrintf -> LogInfo inlining here in a scripted diff 

#29641.

Commit:
log: Remove function name from init logs

It is redundant with -logsourcelocations and the log messages are
clearer without it.

Also, remove a double-space.

Also, add braces around `if` touched in the next commit.

This tiny behavior change requires a test fixup.

fanquake reviewed (commented):
ACK 26a43ed0cf6577bfa6e60b979179b0d99c2ad7db

l0rinc reviewed (approved):
ACK fa4002b3534fbd22a43b32cb24420652d12da6d2

Commit:
log: [refactor] Use info level for init logs

This refactor does not change behavior.

stickies-v commented:
re-ACK face8123fdc10549676c6679ee3225c178a7f30c

fanquake commented:
ACK face8123fdc10549676c6679ee3225c178a7f30c

l0rinc commented:
Post-merge ACK

It seems the post-merge build was partially cancelled for some reason, but the follow-ups passed fully 👍
